os/drivers/testcase/test_signal.c

```diff
@@ -0,0 +1,70 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <errno.h>
+#include <debug.h>
+#include <tinyara/testcase_drv.h>
+#include <tinyara/sched.h>
+#include <signal/signal.h>
+#include "kernel_test_proto.h"
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+int test_get_sig_findaction_add(int cmd, unsigned long arg)
+{
+	FAR sigactq_t *sigact;
+	sigact = sig_findaction(sched_self(), (int)arg);
+	return (int)sigact;
+}
+
+int test_signal_pause(int cmd, unsigned long arg)
+{
+	int ret;
+	ret = pause();              /* pause() always return -1 */
+	if (ret == ERROR && get_errno() == EINTR) {
+		return OK;
+	} else {
+		return ERROR;
+	}
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int test_signal(int cmd, unsigned long arg)
+{
+	switch (cmd) {
+		case TESTIOC_GET_SIG_FINDACTION_ADD: {
+			return test_get_sig_findaction_add(cmd, arg);
+		}
+		break;
+
+		case TESTIOC_SIGNAL_PAUSE: {
+			return test_signal_pause(cmd, arg);
+		}
+		break;
+	}
+}
```

Comment:
The return type of this function is int. Please add return statement.

Reply:
modify